From: Eike Rathke <erack@redhat.com>
Date: Tue, 25 Nov 2014 19:02:08 +0100
Subject: [PATCH] fdo#86708 paint after Undo of Sort

Regression of 10fc138307afb4b39baddb0d56eb8e986e5d29ea that exchanged
the PostPaint with SetDirty, but the paint is also needed.

Change-Id: Ic3e81fa106f405568effaed71ab33b575b4c5c5c
(cherry picked from commit 5cec4ea827570a5bb0a368025f3733b841107d07)
Reviewed-on: https://gerrit.libreoffice.org/13123
Reviewed-by: Andras Timar <andras.timar@collabora.com>
Tested-by: Andras Timar <andras.timar@collabora.com>
---
 sc/source/ui/undo/undosort.cxx | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/sc/source/ui/undo/undosort.cxx b/sc/source/ui/undo/undosort.cxx
--- a/sc/source/ui/undo/undosort.cxx
+++ b/sc/source/ui/undo/undosort.cxx
@@ -49,6 +49,9 @@ void UndoSort::Execute( bool bUndo )
     rDoc.SetDirty(maParam.maSortRange);
     if (!aParam.mbUpdateRefs)
         rDoc.BroadcastCells(aParam.maSortRange, SC_HINT_DATACHANGED);
+
+    pDocShell->PostPaint(maParam.maSortRange, PAINT_GRID);
+    pDocShell->PostDataChanged();
 }
 
 }
-- 
1.9.1

